cli-toolchain/cognito-triggers.md

```diff
@@ -0,0 +1,508 @@
+# Cognito Lambda Triggers
+
+Certain AWS Services can [invoke Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-services.html) in response to lifecycle events. The Amplify CLI provides trigger templates for common use cases.
+
+If you wish to modify the functionality of these templates, you are able to do so locally before pushing them.
+
+Amazon Cognito allows you to set up one Lambda trigger for certain events.  In order to create additional flexibility when configuring Cognito triggers via the CLI, the CLI will create an index file which loops through JavaScript modules.  Each template that you configure is its own JavaScript module.  
+
+You have the opportunity to edit both the index file as well as each module.  For example, when creating a email blacklist [PreSignUp](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-sign-up.html) trigger, you will be asked
```

Comment:
We need to explain what this index file is and why it's necessary. Just a single sentence.

Reply:
The preceeding paragraph attempts to do this: 

> Amazon Cognito allows you to set up one Lambda trigger for certain events.  In order to create additional flexibility when configuring Cognito triggers via the CLI, the CLI will create an index file which loops through JavaScript modules.  Each template that you configure is its own JavaScript module.  

This is more than a single sentence, but it was as concise as I could make it.  Do we need to cut it down?

Reply:
I think I was looking for something like "...this allows you to attach multiple use cases and logical flows to a single lifecycle event" or something like that.